### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by LanskayaIrina**

### CODE REVIEW

File: Ira/my-app-video/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            <Route path={BASE_URL} exact component={Home}/>
+            <Route path={SEARCH}
+                   component={Home}
+            />
+            <Route path={FILM}
+                   render={(
+                     props: RouteComponentProps<{ id: string }>
+                   ) => <FilmPage
+                     {...props}
+                   />}
```
Comment (brusni4ka) [inline Ira/my-app-video/src/App.tsx]: render => component={FilmPage} 
I already asked you to fix it

File: Ira/my-app-video/src/components/filmCard/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const mapDispatchToProps = {
+  fetchMoviesById: fetchMovieById,
+}
+
+const connector = connect(null, mapDispatchToProps);
+
+export type filmCardConnectProps = ConnectedProps<typeof connector>;
+
+export default connector(FilmCard);
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/filmCard/index.ts]: better to use useSelector

File: Ira/my-app-video/src/components/filmPage/FilmPage.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    initPage();
+
+    return () => {
+      resetMovies();
+    }
+  }, []);
+
+  useEffect(() => {
+    initPage();
+  }, [location]);
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/filmPage/FilmPage.tsx]: initPage will be called two times in a row. you can remove second use Effect and add location dependancy in the first one

File: Ira/my-app-video/src/components/home/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  useEffect(() => {
+    getMovies();
+    return () => {
+      dispatch(resetMovies());
+    }
+  }, []);
+
+  useEffect(() => {
+    getMovies();
+  }, [location]);
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/home/Home.tsx]: leave one useEffect. same problem as in Films page

File: Ira/my-app-video/src/components/filmPage/FilmPage.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      genre = currentFilm.genres[0].toLocaleLowerCase();
+    }
+
+    const newParams: string = queryString.stringify({page: btnId});
+    const payLoadParams: { search: string, searchBy: string, sortBy: string, page: number } = {
+      search: genre, searchBy: SearchBy.genre, sortBy: SortBy.release, page: btnId || params.page
+    };
+
+    history.push(`/film/${Number(id)}?${newParams}`);
+    dispatch(fetchMovies(payLoadParams));
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/filmPage/FilmPage.tsx]: useEffect that subscribed on location will be called when you do history.push. You have an issue in logic

File: Ira/my-app-video/src/components/filmPage/FilmPage.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      </div>
+      <div className="top-header-section">
+        <div className="search-result-content">
+          <FullInfoFilmCard film={currentFilm}/>
+        </div>
+      </div>
+      {loading ? <img className="loader film-page-loader" src={loader} alt="loader"/> :
+        <FilmList
+          films={movies}
+        />
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/filmPage/FilmPage.tsx:63]: better to move "Loading..." inside of FilmList component

File: Ira/my-app-video/src/components/home/Home.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+  const onSearchSubmit = ({searchValue, searchBy}: { searchValue: string, searchBy: string }): void => {
+    const newParams: string = queryString.stringify({
+      ...params,
+      search: searchValue,
+      searchBy: searchBy,
+      page: 1,
+    });
+
+    history.push(`/search?${newParams}`);
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/home/Home.tsx:52]: get movies will be called on history.push. Logic issue

File: Ira/my-app-video/src/components/home/Home.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      sortBy: btnName,
+      page: 1,
+    }))
+  };
+
+  const onPage = (btnId: number): void => {
+    const newParams: string = queryString.stringify({...params, sortBy: params.sortBy || 'release_date', page: btnId});
+
+    history.push(`/search?${newParams}`);
+    dispatch(fetchMovies({...params, sortBy: params.sortBy || 'release_date', page: btnId}))
```
Comment (brusni4ka) [inline Ira/my-app-video/src/components/home/Home.tsx]: Why you still using strings like 'release_date'  if you have an ENUM ???

**Review by brusni4ka — CHANGES_REQUESTED** [review]

File: Ira/my-app-video/src/App.tsx (hunk 1 above)
Comment (LanskayaIrina) [reply Ira/my-app-video/src/App.tsx]: I need to pass props

**Review by brusni4ka — APPROVED** [review]

**Pull request merged**